src/metabase/events/view_log.clj

```diff
@@ -12,16 +13,41 @@
    [steffan-westcott.clj-otel.api.trace.span :as span]
    [toucan2.core :as t2]))
 
-(defn increment-view-counts!
-  "Increments the view_count column for a model given a list of ids.
-   Assumes the model has one primary key `id`, and the column for the view count is named `view_count`"
-  [_model & _ids]
-  ;; Disable view_count updates to handle perf issues  (for now) (#44359)
-  #_
-  (when (seq ids)
-    (t2/query {:update (t2/table-name model)
-               :set    {:view_count [:+ :view_count [:inline 1]]}
-               :where  [:in :id ids]})))
+(defn- group-by-frequency
+  [ids]
+  (as-> ids items
+    (seq (frequencies items))
+    (group-by second items)
+    (update-vals items #(map first %))))
+
+(defn- increment-view-counts!*
+  [items]
+  (log/debugf "Increment view counts of %d items" (count items))
+  (try
+    (let [model->ids (as-> items items
+                       (filter (comp some? :id) items)
+                       (group-by :model items)
+                       (update-vals items #(map :id %)))]
+      (doseq [[model ids] model->ids]
+        (doseq [[cnt ids] (group-by-frequency ids)]
```

Comment:
It would be nice if we could cut down the overhead of N * M distinct queries here. One way this could be accomplished would be to flatten the inner loop by generating a query creating an inline table of `(id, cnt)` pairs for each model to use in a single update query per model.

I'm wary of doing too much optimization without quantification though, in particular it's uncertain what the distribution of these items will actually look like in practice. With 5 min window at least it's possible that there is a wide range of frequencies present.

Perhaps it's worth just mentioning that the inner loop could be removed, and how, in case we find this is emitting a burst of too many queries. 

A more pedestrian way to blunt the onslaught could be to put some pauses between each query. Lowering the batch interval could also help spread the mutations a bit more over time.

Reply:
I see you did something similar with a `case` statement for the timestamp update, which is bit easier to express than the table literal. Given how simple it is to define it that way, I'd suggest even more strongly to do this.

Reply:
yea it's much nicer :) 83cdcb6775